### ISSUE TITLE
[Feature]: Allow for lazy loading of each route config

### ISSUE THREAD

**Issue by brandonroberts** [angular/angular#50432]
### Which @angular/* package(s) are relevant/related to the feature request?

router

### Description

Currently, you can eagerly load an entire route config, but if you include providers, guards, and other tokens they will be included in the main bundle even for a lazy loaded component/routes.

```ts
[
  {
    path: 'page',
    canActivate: [authGuardFunction, permissionsGuardFunction],
    loadChldren: () => import('features/page')
  },
  {
    path: 'other-page',
    canActivate: [authGuardFunction, permissionsGuardFunction],
    loadComponent: () => import('about.component')
  }
]
```

This feature would allow you to provide a route config with a `path` and a `loadConfig/loadRoute` that resolves to the rest of the route configuration, and would be resolved on the matching of that path before proceeding. This would reduce the main bundle size for route-level configuration, while allowing the entire routing tree to be defined eagerly.

### Proposed solution

Allow lazy loading of the entire route config except for the path

```ts
// features/page.ts
const config: RouteConfig =  {
  canActivate: [authGuardFunction, permissionsGuardFunction],
  component: FeatureHome,
  children: [
    // more lazy loaded routes
  ]
}

export default config;
```

```ts
[
  {
    path: 'page',
    loadConfig: () => import('features/page')
    // or
    loadRoute: () => import('features/page')
  }
]
```

### Alternatives considered

In Analog, file-based routing is used along with `loadChildren` to lazy load the entire route config

```ts
[
  {
    path: 'page',
    loadChildren: () => import('features/page').then(m => {
      return [
       {
          path: '',
          component: m.default,
          ...m.routeMeta
       }
      ];
    })
  }
]
```

**Comment by alxhub** [comment]
You can do this by using an empty-path route:

```typescript
export default [{
  path: '',
  pathMatch: 'prefix', // I don't think this is actually necessary
  providers: [...],
  canActivate: [authGuardFunction, permissionsGuardFunction],
  children: [{path: 'page', ...}],
}];
```

And then loading that config via`loadChildren`.

**Comment by atscott** [comment]
@alxhub is correct, this can be accomplished pretty effectively with an empty path route. The only difference between that and the `loadConfig`/`loadRoute` request here would be the removal of that empty path property in the loaded route.

**Comment by brandonroberts** [comment]
I see. Would this cause the router to eagerly load each top-level route with an empty path and `loadChildren` until it finds a match to traverse down further?

**Comment by alxhub** [comment]
No, because the path (`page` in your example) is determined by the outer (eager) config - the lazy loading is only triggered when the path is matched. The loaded config uses the empty path to group a bunch of child routes and attach providers/guards/etc.

**Comment by markostanimirovic** [comment]
Hey @alxhub @AndrewKushnir @atscott 👋 

We've faced an issue caused by nested `loadChildren`: https://github.com/analogjs/analog/issues/536

The `loadConfig` option would probably solve that.
